redland/docs/redland.3
.\" Automatically generated by Pod::Man v1.34, Pod::Parser v1.13
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sh \" Subsection heading
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  | will give a
.\" real vertical bar.  \*(C+ will give a nicer C++.  Capital omega is used to
.\" do unbreakable dashes and therefore won't be available.  \*(C` and \*(C'
.\" expand to `' in nroff, nothing in troff, for use with C<>.
.tr \(*W-|\(bv\*(Tr
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.Sh), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.if \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.\"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.hy 0
.if n .na
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "redland 3"
.TH redland 3 "redland 0.9.16" "23-Feb-2004" "Redland RDF Application Framework"
.SH "NAME"
Redland Resource Description Framework (RDF) Library
.SH "VERSION"
.IX Header "VERSION"
\&\s-10.9.16\s0
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
.Vb 1
\& #include <redland.h>
.Ve
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
\&\fBredland\fR is a library providing support for the
Resource Description Framework (\s-1RDF\s0) written in \s-1ANSI\s0 C with
APIs in several other languages.
.PP
This manual page lists most of the redland 0.9.15 functions for the C
\&\s-1API\s0 but does not claim to be a complete summary of the entire \s-1API\s0 at
this time.  For the complete \s-1API\s0 with full details of the function
interface, see the \s-1HTML\s0 \s-1API\s0 documentation either on the Redland web
site at <http://www.redland.opensource.ac.uk/> or with the software
release in the docs/api directory.  Documentation on the Perl and
Python APIs in \s-1POD\s0 and Pydoc respectively are also available.
.SH "FUNCTIONS"
.IX Header "FUNCTIONS"
The functions defined by \fBredland\fR are all defined with the
\&\f(CW\*(C`librdf_\*(C'\fR prefix
.Sh "class world"
.IX Subsection "class world"
.IP "librdf_world* \fBlibrdf_new_world\fR(\fIvoid\fR)" 4
.IX Item "librdf_world* librdf_new_world(void)"
.PD 0
.IP "void \fBlibrdf_free_world\fR(librdf_world* \fIworld\fR)" 4
.IX Item "void librdf_free_world(librdf_world* world)"
.IP "void \fBlibrdf_world_open\fR(librdf_world* \fIworld\fR)" 4
.IX Item "void librdf_world_open(librdf_world* world)"
.IP "void \fBlibrdf_world_set_error\fR(librdf_world* \fIworld\fR, void* \fIuser_data\fR, void (*\fIerror_fn\fR)(void* \fIuser_data\fR, const char* \fImsg\fR, ...))" 4
.IX Item "void librdf_world_set_error(librdf_world* world, void* user_data, void (*error_fn)(void* user_data, const char* msg, ...))"
.IP "void \fBlibrdf_world_set_warning\fR(librdf_world* \fIworld\fR, void* \fIuser_data\fR, void (*\fIwarning_fn\fR)(void* \fIuser_data\fR, const char* \fImsg\fR, ...))" 4
.IX Item "void librdf_world_set_warning(librdf_world* world, void* user_data, void (*warning_fn)(void* user_data, const char* msg, ...))"
.IP "void \fBlibrdf_world_set_digest\fR(librdf_world*, const char* \fIname\fR)" 4
.IX Item "void librdf_world_set_digest(librdf_world*, const char* name)"
.IP "void \fBlibrdf_world_set_uris_hash\fR(librdf_world* \fIworld\fR, librdf_hash* \fIuris_hash\fR)" 4
.IX Item "void librdf_world_set_uris_hash(librdf_world* world, librdf_hash* uris_hash)"
.IP "const char* \fBlibrdf_world_get_feature\fR(librdf_world* \fIworld\fR, librdf_uri* \fIfeature\fR)" 4
.IX Item "const char* librdf_world_get_feature(librdf_world* world, librdf_uri* feature)"
.IP "int \fBlibrdf_world_set_feature\fR(librdf_world* \fIworld\fR, librdf_uri* \fIfeature\fR, const char* \fIvalue\fR)" 4
.IX Item "int librdf_world_set_feature(librdf_world* world, librdf_uri* feature, const char* value)"
.PD
.Sh "class iterator"
.IX Subsection "class iterator"
.IP "librdf_iterator* \fBlibrdf_new_iterator\fR(librdf_world* \fIworld\fR, void* \fIcontext\fR, int (*\fIis_end\fR)(void*), void* (*\fIget_next\fR)(void*), void (*\fIfinished\fR)(void*))" 4
.IX Item "librdf_iterator* librdf_new_iterator(librdf_world* world, void* context, int (*is_end)(void*), void* (*get_next)(void*), void (*finished)(void*))"
.PD 0
.IP "void \fBlibrdf_free_iterator\fR(librdf_iterator*)" 4
.IX Item "void librdf_free_iterator(librdf_iterator*)"
.IP "int \fBlibrdf_iterator_end\fR(librdf_iterator* \fIiterator\fR)" 4
.IX Item "int librdf_iterator_end(librdf_iterator* iterator)"
.IP "int \fBlibrdf_iterator_finished\fR(librdf_iterator* \fIiterator\fR)" 4
.IX Item "int librdf_iterator_finished(librdf_iterator* iterator)"
.IP "int \fBlibrdf_iterator_next\fR(librdf_iterator* \fIiterator\fR)" 4
.IX Item "int librdf_iterator_next(librdf_iterator* iterator)"
.IP "void* \fBlibrdf_iterator_get_object\fR(librdf_iterator* \fIiterator\fR)" 4
.IX Item "void* librdf_iterator_get_object(librdf_iterator* iterator)"
.IP "void* \fBlibrdf_iterator_get_context\fR(librdf_iterator* \fIiterator\fR)" 4
.IX Item "void* librdf_iterator_get_context(librdf_iterator* iterator)"
.IP "void* \fBlibrdf_iterator_get_key\fR(librdf_iterator* \fIiterator\fR)" 4
.IX Item "void* librdf_iterator_get_key(librdf_iterator* iterator)"
.IP "void* \fBlibrdf_iterator_get_value\fR(librdf_iterator* \fIiterator\fR)" 4
.IX Item "void* librdf_iterator_get_value(librdf_iterator* iterator)"
.IP "int \fBlibrdf_iterator_add_map\fR(librdf_iterator* \fIiterator\fR, void* (*\fIfn\fR)(void* \fIcontext\fR, void* \fIitem\fR), void* \fIcontext\fR)" 4
.IX Item "int librdf_iterator_add_map(librdf_iterator* iterator, void* (*fn)(void* context, void* item), void* context)"
.IP "void* \fBlibrdf_iterator_map_remove_duplicate_nodes\fR(void* \fIitem\fR, void* \fIuser_data\fR)" 4
.IX Item "void* librdf_iterator_map_remove_duplicate_nodes(void* item, void* user_data)"
.PD
.Sh "class digest"
.IX Subsection "class digest"
.IP "void \fBlibrdf_digest_register_factory\fR(librdf_world* \fIworld\fR, const char* \fIname\fR, void (*\fIfactory\fR) (librdf_digest_factory*))" 4
.IX Item "void librdf_digest_register_factory(librdf_world* world, const char* name, void (*factory) (librdf_digest_factory*))"
.PD 0
.IP "librdf_digest_factory* \fBlibrdf_get_digest_factory\fR(librdf_world* \fIworld\fR, const char* \fIname\fR)" 4
.IX Item "librdf_digest_factory* librdf_get_digest_factory(librdf_world* world, const char* name)"
.IP "librdf_digest* \fBlibrdf_new_digest\fR(librdf_world* \fIworld\fR, char* \fIname\fR)" 4
.IX Item "librdf_digest* librdf_new_digest(librdf_world* world, char* name)"
.IP "librdf_digest* \fBlibrdf_new_digest_from_factory\fR(librdf_world* \fIworld\fR, librdf_digest_factory* \fIfactory\fR)" 4
.IX Item "librdf_digest* librdf_new_digest_from_factory(librdf_world* world, librdf_digest_factory* factory)"
.IP "void \fBlibrdf_free_digest\fR(librdf_digest* \fIdigest\fR)" 4
.IX Item "void librdf_free_digest(librdf_digest* digest)"
.IP "void \fBlibrdf_digest_init\fR(librdf_digest* \fIdigest\fR)" 4
.IX Item "void librdf_digest_init(librdf_digest* digest)"
.IP "void \fBlibrdf_digest_update\fR(librdf_digest* \fIdigest\fR, unsigned char* \fIbuf\fR, size_t \fIlength\fR)" 4
.IX Item "void librdf_digest_update(librdf_digest* digest, unsigned char* buf, size_t length)"
.IP "void \fBlibrdf_digest_final\fR(librdf_digest* \fIdigest\fR)" 4
.IX Item "void librdf_digest_final(librdf_digest* digest)"
.IP "void* \fBlibrdf_digest_get_digest\fR(librdf_digest* \fIdigest\fR)" 4
.IX Item "void* librdf_digest_get_digest(librdf_digest* digest)"
.IP "char* \fBlibrdf_digest_to_string\fR(librdf_digest* \fIdigest\fR)" 4
.IX Item "char* librdf_digest_to_string(librdf_digest* digest)"
.IP "void \fBlibrdf_digest_print\fR(librdf_digest* \fIdigest\fR, FILE* \fIfh\fR)" 4
.IX Item "void librdf_digest_print(librdf_digest* digest, FILE* fh)"
.PD
.Sh "class uri"
.IX Subsection "class uri"
.IP "librdf_uri* \fBlibrdf_new_uri\fR(librdf_world* \fIworld\fR, const unsigned char * \fIstring\fR)" 4
.IX Item "librdf_uri* librdf_new_uri(librdf_world* world, const unsigned char * string)"
.PD 0
.IP "librdf_uri* \fBlibrdf_new_uri_from_uri\fR(librdf_uri* \fIuri\fR)" 4
.IX Item "librdf_uri* librdf_new_uri_from_uri(librdf_uri* uri)"
.IP "librdf_uri* \fBlibrdf_new_uri_from_uri_local_name\fR(librdf_uri* \fIuri\fR, const unsigned char* \fIlocal_name\fR)" 4
.IX Item "librdf_uri* librdf_new_uri_from_uri_local_name(librdf_uri* uri, const unsigned char* local_name)"
.IP "void \fBlibrdf_free_uri\fR(librdf_uri* \fIuri\fR)" 4
.IX Item "void librdf_free_uri(librdf_uri* uri)"
.IP "unsigned char* \fBlibrdf_uri_as_string\fR(librdf_uri* \fIuri\fR)" 4
.IX Item "unsigned char* librdf_uri_as_string(librdf_uri* uri)"
.IP "unsigned char* \fBlibrdf_uri_as_counted_string\fR(librdf_uri* \fIuri\fR, size_t* \fIlen_p\fR)" 4
.IX Item "unsigned char* librdf_uri_as_counted_string(librdf_uri* uri, size_t* len_p)"
.IP "librdf_digest* \fBlibrdf_uri_get_digest\fR(librdf_uri* \fIuri\fR)" 4
.IX Item "librdf_digest* librdf_uri_get_digest(librdf_uri* uri)"
.IP "void librdf_uri_print>(librdf_uri* \fIuri\fR, FILE* \fIfh\fR)" 4
.IX Item "void librdf_uri_print>(librdf_uri* uri, FILE* fh)"
.IP "unsigned char* \fBlibrdf_uri_to_string\fR(librdf_uri* \fIuri\fR)" 4
.IX Item "unsigned char* librdf_uri_to_string(librdf_uri* uri)"
.IP "unsigned char* \fBlibrdf_uri_to_counted_string\fR(librdf_uri* \fIuri\fR, size_t* \fIlen_p\fR)" 4
.IX Item "unsigned char* librdf_uri_to_counted_string(librdf_uri* uri, size_t* len_p)"
.IP "int \fBlibrdf_uri_equals\fR(librdf_uri* \fIfirst_uri\fR, librdf_uri* \fIsecond_uri\fR)" 4
.IX Item "int librdf_uri_equals(librdf_uri* first_uri, librdf_uri* second_uri)"
.IP "int \fBlibrdf_uri_is_file_uri\fR(librdf_uri* \fIuri\fR)" 4
.IX Item "int librdf_uri_is_file_uri(librdf_uri* uri)"
.IP "const char* \fBlibrdf_uri_to_filename\fR(librdf_uri* \fIuri\fR)" 4
.IX Item "const char* librdf_uri_to_filename(librdf_uri* uri)"
.IP "librdf_uri* \fBlibrdf_new_uri_normalised_to_base\fR(const unsigned char* \fIuri_string\fR, librdf_uri* \fIsource_uri\fR, librdf_uri* \fIbase_uri\fR)" 4
.IX Item "librdf_uri* librdf_new_uri_normalised_to_base(const unsigned char* uri_string, librdf_uri* source_uri, librdf_uri* base_uri)"
.IP "librdf_uri* \fBlibrdf_new_uri_relative_to_base\fR(librdf_uri* \fIbase_uri\fR, const unsigned char* \fIuri_string\fR)" 4
.IX Item "librdf_uri* librdf_new_uri_relative_to_base(librdf_uri* base_uri, const unsigned char* uri_string)"
.IP "librdf_uri* \fBlibrdf_new_uri_from_filename\fR(librdf_world* \fIworld\fR, const char* \fIfilename\fR)" 4
.IX Item "librdf_uri* librdf_new_uri_from_filename(librdf_world* world, const char* filename)"
.PD
.Sh "class node"
.IX Subsection "class node"
.IP "librdf_node* \fBlibrdf_new_node\fR(librdf_world* \fIworld\fR)" 4
.IX Item "librdf_node* librdf_new_node(librdf_world* world)"
.PD 0
.IP "librdf_node* \fBlibrdf_new_node_from_uri_string\fR(librdf_world* \fIworld\fR, const char* \fIstring\fR)" 4
.IX Item "librdf_node* librdf_new_node_from_uri_string(librdf_world* world, const char* string)"
.IP "librdf_node* \fBlibrdf_new_node_from_uri\fR(librdf_world* \fIworld\fR, librdf_uri* \fIuri\fR)" 4
.IX Item "librdf_node* librdf_new_node_from_uri(librdf_world* world, librdf_uri* uri)"
.IP "librdf_node* \fBlibrdf_new_node_from_uri_local_name\fR(librdf_world* \fIworld\fR, librdf_uri* \fIuri\fR, const char* \fIlocal_name\fR)" 4
.IX Item "librdf_node* librdf_new_node_from_uri_local_name(librdf_world* world, librdf_uri* uri, const char* local_name)"
.IP "librdf_node* \fBlibrdf_new_node_from_normalised_uri_string\fR(librdf_world* \fIworld\fR, const char* \fIuri_string\fR, librdf_uri* \fIsource_uri\fR, librdf_uri* \fIbase_uri\fR)" 4
.IX Item "librdf_node* librdf_new_node_from_normalised_uri_string(librdf_world* world, const char* uri_string, librdf_uri* source_uri, librdf_uri* base_uri)"
.IP "librdf_node* \fBlibrdf_new_node_from_literal\fR(librdf_world* \fIworld\fR, const char* \fIstring\fR, const char* \fIxml_language\fR, int \fIxml_space\fR, int \fIis_wf_xml\fR)" 4
.IX Item "librdf_node* librdf_new_node_from_literal(librdf_world* world, const char* string, const char* xml_language, int xml_space, int is_wf_xml)"
.IP "librdf_node* \fBlibrdf_new_node_from_typed_literal\fR(librdf_world* \fIworld\fR, const unsigned char* \fIstring\fR, const char* \fIxml_language\fR, librdf_uri* \fIdatatype_uri\fR)" 4
.IX Item "librdf_node* librdf_new_node_from_typed_literal(librdf_world* world, const unsigned char* string, const char* xml_language, librdf_uri* datatype_uri)"
.IP "librdf_node* \fBlibrdf_new_node_from_blank_identifier\fR(librdf_world* \fIworld\fR, const unsigned char* \fIidentifier\fR)" 4
.IX Item "librdf_node* librdf_new_node_from_blank_identifier(librdf_world* world, const unsigned char* identifier)"
.IP "librdf_node* \fBlibrdf_new_node_from_node\fR(librdf_node* \fInode\fR)" 4
.IX Item "librdf_node* librdf_new_node_from_node(librdf_node* node)"
.IP "void \fBlibrdf_node_init\fR(librdf_world* \fIworld\fR, librdf_node* \fInode\fR)" 4
.IX Item "void librdf_node_init(librdf_world* world, librdf_node* node)"
.IP "void \fBlibrdf_free_node\fR(librdf_node* \fIr\fR)" 4
.IX Item "void librdf_free_node(librdf_node* r)"
.IP "librdf_uri* \fBlibrdf_node_get_uri\fR(librdf_node* \fInode\fR)" 4
.IX Item "librdf_uri* librdf_node_get_uri(librdf_node* node)"
.IP "librdf_node_type \fBlibrdf_node_get_type\fR(librdf_node* \fInode\fR)" 4
.IX Item "librdf_node_type librdf_node_get_type(librdf_node* node)"
.IP "unsigned char* \fBlibrdf_node_get_literal_value\fR(librdf_node* \fInode\fR)" 4
.IX Item "unsigned char* librdf_node_get_literal_value(librdf_node* node)"
.IP "unsigned char* \fBlibrdf_node_get_literal_value_as_counted_string\fR(librdf_node* \fInode\fR, size_t* \fIlen_p\fR)" 4
.IX Item "unsigned char* librdf_node_get_literal_value_as_counted_string(librdf_node* node, size_t* len_p)"
.IP "char* \fBlibrdf_node_get_literal_value_as_latin1\fR(librdf_node* \fInode\fR)" 4
.IX Item "char* librdf_node_get_literal_value_as_latin1(librdf_node* node)"
.IP "char* \fBlibrdf_node_get_literal_value_language\fR(librdf_node* \fInode\fR)" 4
.IX Item "char* librdf_node_get_literal_value_language(librdf_node* node)"
.IP "int \fBlibrdf_node_get_literal_value_is_wf_xml\fR(librdf_node* \fInode\fR)" 4
.IX Item "int librdf_node_get_literal_value_is_wf_xml(librdf_node* node)"
.IP "librdf_uri* \fBlibrdf_node_get_literal_value_datatype_uri\fR(librdf_node* \fInode\fR)" 4
.IX Item "librdf_uri* librdf_node_get_literal_value_datatype_uri(librdf_node* node)"
.IP "int \fBlibrdf_node_get_li_ordinal\fR(librdf_node* \fInode\fR)" 4
.IX Item "int librdf_node_get_li_ordinal(librdf_node* node)"
.IP "unsigned char* \fBlibrdf_node_get_blank_identifier\fR(librdf_node* \fInode\fR)" 4
.IX Item "unsigned char* librdf_node_get_blank_identifier(librdf_node* node)"
.IP "int \fBlibrdf_node_is_resource\fR(librdf_node* \fInode\fR)" 4
.IX Item "int librdf_node_is_resource(librdf_node* node)"
.IP "int \fBlibrdf_node_is_literal\fR(librdf_node* \fInode\fR)" 4
.IX Item "int librdf_node_is_literal(librdf_node* node)"
.IP "int \fBlibrdf_node_is_blank\fR(librdf_node* \fInode\fR)" 4
.IX Item "int librdf_node_is_blank(librdf_node* node)"
.IP "librdf_digest* \fBlibrdf_node_get_digest\fR(librdf_node* \fInode\fR)" 4
.IX Item "librdf_digest* librdf_node_get_digest(librdf_node* node)"
.IP "size_t \fBlibrdf_node_encode\fR(librdf_node* \fInode\fR, unsigned char* \fIbuffer\fR, size_t \fIlength\fR)" 4
.IX Item "size_t librdf_node_encode(librdf_node* node, unsigned char* buffer, size_t length)"
.IP "size_t \fBlibrdf_node_decode\fR(librdf_node* \fInode\fR, unsigned char* \fIbuffer\fR, size_t \fIlength\fR)" 4
.IX Item "size_t librdf_node_decode(librdf_node* node, unsigned char* buffer, size_t length)"
.IP "unsigned char* \fBlibrdf_node_to_string\fR(librdf_node* \fInode\fR)" 4
.IX Item "unsigned char* librdf_node_to_string(librdf_node* node)"
.IP "unsigned char* \fBlibrdf_node_to_counted_string\fR(librdf_node* \fInode\fR, size_t* \fIlen_p\fR)" 4
.IX Item "unsigned char* librdf_node_to_counted_string(librdf_node* node, size_t* len_p)"

redland/NEWS

                   Redland RDF Application Framework - News

2004-02-24 Redland 0.9.16 Released

   A  small  but  important  bug fix to make the installed C headers work
   correctly again.

   See the [1]0.9.16 Release Notes for the full details of the changes.

2004-01-31 Redland 0.9.15 Released

   API  CHANGE  Several  classes  were modified to use unsigned char* for
   UTF-8 encoded URIs and Literals
   Update  to  synchronise  with  [2]Raptor  version  1.2.0  (RDF/XML and
   N-Triples parser) - see [3]Raptor NEWS gaining the [4]Turtle Terse RDF
   Triple Language
   Added storage with MySQL written by [5]Morten Frederiksen
   Added storage with [6]AKT Triplestore
   Added file and URI-backed in-memory storage
   Added  fixes  to  make  Redland compile cleanly under C++ (tested with
   g++)
   Added assertion checks and reporting for function arguments
   Added method for finding statements in a context
   Added a method for listing the contexts in a graph
   Storage,  model  and  serializer  get/set feature methods now take and
   return nodes
   Improved  rdfproc  utility  to  handle multiple storage types, storage
   options, contexts and environment variables for configuration.
   Fix crash on reporting parser errors with % in the message
   Improved RDF/XML serializer to chose better output names
   Many other minor changes and fixes.

   See  the  [7]Detailed  0.9.15  Release  Notes for the full list of the
   changes.

2003-09-08 Redland 0.9.14 Released

   Update  to  synchronise  with  [8]Raptor  version  1.0.0  (RDF/XML and
   N-Triples parser) - see [9]Raptor NEWS for the changes since 0.9.12.
   Added  a  new  utility  program  rdfproc - a general RDF processor for
   Redland that allows command line use of the library.
   Made the Node class intern every node.
   Fixed the broken use of Perl UNIVERSAL::isa
   Other minor API changes

   See  the  [10]Detailed  0.9.14  Release Notes for the full list of the
   changes.

2003-08-28 Redland 0.9.13 Released

   API  CHANGE  [11]Node The class was modifed to make objects immutable,
   removing the set methods and to add new node type convienence methods.
   API addition [12]Model Added a method to sync the model to the backing
   store.
   API addition [13]Parser Added methods to allow parsing from strings.
   API   addition   [14]Statement  Added  method  to  check  for  partial
   statements.
   API addition [15]Storage Added a method to sync the backing store.
   Better error recovery - most errors return if possible.
   Added initial simple threading support (patches by Seth Ladd)
   Greatly  improved  Python API using 2.2 features (Edd Dumbill and Matt
   Biddulph)
   Updated  Perl,  Python  and  Java  APIs to always return new nodes and
   statements, removing the need to consider shared object issues.
   Updated serializing to RDF/XML to handle datatype literals
   Update  to  [16]Raptor version 0.9.12 (RDF/XML and N-Triples parser) -
   see [17]Raptor NEWS for changes since 0.9.8.
   Extensive portability and other bug fixes.

   See  the  [18]Detailed  0.9.13  Release Notes for the full list of the
   changes.

2003-02-13 Redland 0.9.12 Released

   FORMAT CHANGE The persistent storage format was changed to support RDF
   typed  literals  and  this  will  require  an  upgrade of any existing
   Berkeley/Sleepycat  DB  stores created by Redland 0.9.11 or earlier. A
   utility  redland-db-upgrade  is provided that creates an updated store
   from an existing one.
   API  CHANGE  [19]Iterator  and  [20]Stream  classes. The next/get_next
   method  is  split into get_object always returning a shared object and
   next  methods. get_context method added. (get_object is called current
   in higher level language APIs)
   API  CHANGE  [21]Model  class.  Method  add_statement  no longer takes
   ownership   of  the  passed  in  statement.  The  caller  now  retains
   ownership.
   API  CHANGE [22]Model class. Method add_statements no longer frees the
   passed in librdf_stream; the caller must free it now.
   API  CHANGE  [23]Node  and  [24]Model  classes.  Added  RDF  Datatyped
   Literals. Other literal is_wf_xml constructors and methods deprecated.
   API  CHANGE  All  literals methods and constructors have no mention of
   the never-used or supported XML Space argument which is not part of an
   RDF literal.
   Added  a  script [25]utils/update-api-0912.pl to help automate the API
   changes  as  far  as  possible  or  warn  about  those  that cannot be
   automatically updated.
   Added  contexts  - Statements added to [26]Model (when using a Storage
   that supports contexts) can take an optional [27]Node, returned by the
   new get_context method on [28]Iterator and [29]Stream classes.
   Many  portability  and compiling fixes (OSX, gcc 3.x, FreeBSD, non-gcc
   compiler)
   [30]Java   API   classes.  Add  finished()  method  replacing  useless
   finalize().
   Use Raptor message callbacks to get parser warnings and errors
   Build  Java  API working around the stupid things recent SWIG versions
   do.
   Make Redland work with BDB/Sleepycat DB v4 (as well as v2 and v3)
   [31]Python API gains [32]pydoc comments and HTML derived version
   [33]Perl and [34]Python APIs now receive Redland message callbacks
   Redland  URI  class  now  registers  with Raptor rather than specially
   compile Raptor
   Added RDF/XML [35]Serializer class and made higher level language APIs
   to it
   Update  to  [36]Raptor  version 0.9.8 (RDF/XML and N-Triples parser) -
   see [37]Raptor NEWS for detailed changes since 0.9.5
   Added skeleton [38]Ruby and [39]PHP APIs
   Major source reorganisation.

   See also the [40]Detailed 0.9.12 Release Notes

2002-06-08 Redland 0.9.11 Released

   Update to [41]Raptor version 0.9.5 (RDF/XML and N-Triples parser)
   Added  [42]serialization  to  a  syntax API; with [43]N-Triples as the
   first syntax
   Added start of [44]query language API

redland/perl/lib/RDF/Redland.pm
# -*- Mode: Perl -*-
#
# Redland.pm - Redland top level Perl module
#
# $Id: Redland.pm,v 1.1.1.1 2005/07/06 09:38:37 gully Exp $
#
# Copyright (C) 2000-2001 David Beckett - http://purl.org/net/dajobe/
# Institute for Learning and Research Technology - http://www.ilrt.org/
# University of Bristol - http://www.bristol.ac.uk/
# 
# This package is Free Software or Open Source available under the
# following licenses (these are alternatives):
#   1. GNU Lesser General Public License (LGPL)
#   2. GNU General Public License (GPL)
#   3. Mozilla Public License (MPL)
# 
# See LICENSE.html or LICENSE.txt at the top of this package for the
# full license terms.
# 
# 
#

use RDF::Redland::Iterator;
use RDF::Redland::Model;
use RDF::Redland::Node;
use RDF::Redland::Parser;
use RDF::Redland::Serializer;
use RDF::Redland::Statement;
use RDF::Redland::Storage;
use RDF::Redland::Stream;
use RDF::Redland::URI;

use RDF::Redland::CORE;

package RDF::Redland::World;

sub new ($) {
  my($proto)=@_;
  my $class = ref($proto) || $proto;
  my $self  = {};

  $self->{WORLD}=&RDF::Redland::CORE::librdf_new_world();
  &RDF::Redland::CORE::librdf_world_open($self->{WORLD});

  &RDF::Redland::CORE::librdf_perl_world_init($self->{WORLD});

  bless ($self, $class);
  $self->{ME}=$self;
  return $self;
}

sub DESTROY ($) {
  warn "RDF::World DESTROY\n" if $RDF::Debug;
  &RDF::Redland::CORE::librdf_perl_world_finish();
}

sub message ($$) {
  my($type,$message)=@_;
  if($type == 0) {
    if(ref $RDF::Redland::Error_Sub) {
      $RDF::Redland::Error_Sub->($message);
    } else {
      die "Redland error: $message\n";
    }
  } else {
    if(ref $RDF::Redland::Warning_Sub) {
      $RDF::Redland::Warning_Sub->($message);
    } else {
      warn "Redland warning: $message\n";
    }
  }
}

package RDF::Redland;

use vars qw($Debug $World $Error_Sub $Warning_Sub);

$Debug=0;

$World=new RDF::Redland::World;

$Error_Sub=undef;

$Warning_Sub=undef;

sub set_error_handler ($) {
  $Error_Sub=shift;
}

sub set_warning_handler ($) {
  $Warning_Sub=shift;
}


=pod

=head1 NAME

RDF::Redland - Redland RDF Class

=head1 SYNOPSIS

  use RDF::Redland;
  my $storage=new RDF::Redland::Storage("hashes", "test", "new='yes',hash-type='memory'");
  my $model=new RDF::Redland::Model($storage, "");

  ...

=head1 DESCRIPTION

This class initialises the Redland RDF classes.

=head1 STATIC METHODS

=over

=item set_error_handler SUB

Set I<SUB> as the subroutine to be called on a Redland error with
the error message as the single argument.  For example:

  RDF::Redland::set_error_handler(sub {
    my $msg=shift;
    # Do something with $msg
  });

The default if this is not set, is to run die $msg

=item set_warning_handler SUB

Set I<SUB> as the subroutine to be called on a Redland warning with
the warning message as the single argument.  For example:

  RDF::Redland::set_warning_handler(sub {
    my $msg=shift;
    # Do something with $msg
  });

The default if this is not set, is to run warn $msg

=back

=head1 SEE ALSO

L<RDF::Redland::Node>, L<RDF::Redland::Statement>, L<RDF::Redland::Model>,
L<RDF::Redland::Storage>, L<RDF::Redland::Parser>, L<RDF::Redland::Iterator>,
L<RDF::Redland::Stream>, L<RDF::Redland::URI> and L<RDF::Redland::RSS>.

=head1 AUTHOR

Dave Beckett - http://purl.org/net/dajobe/

=cut

1;
